components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -1023,6 +1030,159 @@ public void updateResidentIdP(IdentityProvider identityProvider, String tenantDo
         int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
         return dao.getIdPsSearch(null, tenantId, tenantDomain, filter);
     }
+
+    /**
+     * Get all basic identity provider information.
+     *
+     * @param limit     limit per page.
+     * @param offset    offset value.
+     * @param filter    filter value for IdP search.
+     * @param sortOrder order of IdP ASC/DESC.
+     * @param sortBy    the column value need to sort.
+     * @return Identity Provider's Basic Information array.
+     * @throws IdentityProviderManagementException Error when getting list of Identity Providers.
+     */
+    @Override
+    public IdpSearchResult getIdPs(int limit, int offset, String filter, String sortOrder, String sortBy,
+            String tenantDomain) throws IdentityProviderManagementException, IOException {
+
+        IdpSearchResult result = new IdpSearchResult();
+        if (limit > 0) {
```

Comment:
lets try to use guard clause pattern, instead of else.
https://medium.com/better-programming/why-you-need-to-stop-using-else-statements-5b1fd09dea9e 

ex:if(limit<0 && limit=0){
result.setLimit(limit);
            result.setOffSet(offset);
            result.setFilter(filter);
            result.setSortOrder(sortOrder);
            result.setSortBy(sortBy);
            result.setIdpCount(0);
            result.setIdpList(null);
return result;
}
            List<ExpressionNode> expressionNodes = getExpressionNodes(filter);
            validateAndSetParameters(limit, offset, sortOrder, sortBy, result);
            int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
            result.setIdpCount(getTotalIdPCount(expressionNodes, tenantId));
            result.setIdpList(dao.getIdPsSearch(tenantId, expressionNodes, result.getLimit(), result.getOffSet(),
                    result.getSortOrder(), result.getSortBy()));
return result;